xblockmufi/xblockmufi.py

```diff
@@ -194,8 +192,18 @@ def student_submit(self, data, suffix=''):
         """
         Save student answer
         """
-        self.student_answer = data['answer']
-        return {'success':True}
+
+        if self.is_past_due():
+            return {
+                'success':False,
+                'button_display_class':self._get_button_display_class(),
+            }
+        else:
+            self.student_answer = data['answer']
+            return {
+                'success':True,
+                'button_display_class': self._get_button_display_class(),
```

Comment:
I like this a lot better, but looking at this we might be able to simplify it further:

```python
if self.is_past_due():
    success_value = False
else:
    success_value = True
    self.student_answer = data['answer']
return {
    'success': success_value,
    'button_display_class': self._get_button_display_class(),
}
```